app/vmselect/promql/transform.go

```diff
@@ -826,6 +826,9 @@ func transformHistogramQuantiles(tfa *transformFuncArg) ([]*timeseries, error) {
 	// Calculate quantile individually per each phi.
 	var rvs []*timeseries
 	for _, phiArg := range phiArgs {
+		if len(phiArg) == 0 || (phiArg[0].Values[0] < 0 || phiArg[0].Values[0] > 1) {
+			return nil, fmt.Errorf("got unexpected phi arg. it should containes only numbers from range [0..1]")
```

Comment:
The problem here is that we can't say what was the original argument here... That's why I think we might want to return an error earlier, where we still have the original value passed to the function.